packages/core/test/input.test.ts

```diff
@@ -159,4 +159,204 @@ describe('input', () => {
+            src: createMachine({
+              context: ({ input }) => {
+                spy(input);
+                return {};
+              }
+            }),
+            input: 42
+          }
+        }
+      }
+    );
+
+    interpret(machine).start();
+
+    expect(spy).toHaveBeenCalledWith(42);
+  });
+
+  it('should provide input to the referenced actor defined together with dynamic input when invoking', () => {
+    const spy = jest.fn();
+
+    const machine = createMachine(
+      {
+        invoke: {
+          src: 'child'
+        }
+      },
+      {
+        actors: {
+          child: {
+            src: createMachine({
+              context: ({ input }) => {
+                spy(input);
+                return {};
+              }
+            }),
+            input: (_, { input }) => input + 100
+          }
+        }
+      }
+    );
+
+    interpret(machine, { input: 42 }).start();
+
+    expect(spy).toHaveBeenCalledWith(142);
+  });
+
+  it('should provide input to the referenced actor defined together with dynamic input when spawning', () => {
+    const spy = jest.fn();
+
+    const machine = createMachine(
+      {
+        entry: assign((_ctx, _ev, { spawn }) => ({
+          childRef: spawn('child')
+        }))
+      },
+      {
+        actors: {
+          child: {
+            src: createMachine({
+              context: ({ input }) => {
+                spy(input);
+                return {};
+              }
+            }),
+            input: (_, { input }) => input + 100
+          }
+        }
+      }
+    );
+
+    interpret(machine, { input: 42 }).start();
+
+    expect(spy).toHaveBeenCalledWith(142);
+  });
+
+  it('should prioritize inline input over the one defined with referenced actor when invoking', () => {
+    const spy = jest.fn();
+
+    const machine = createMachine(
+      {
+        invoke: {
+          src: 'child',
+          input: 69
+        }
+      },
+      {
+        actors: {
+          child: {
+            src: createMachine({
+              context: ({ input }) => {
+                spy(input);
+                return {};
+              }
+            }),
+            input: 42
+          }
+        }
+      }
+    );
+
+    interpret(machine).start();
+
+    expect(spy).toHaveBeenCalledWith(69);
+  });
+
+  it('should prioritize inline input over the one defined with referenced actor when spawning', () => {
+    const spy = jest.fn();
+
+    const machine = createMachine(
+      {
+        entry: assign((_ctx, _ev, { spawn }) => ({
+          childRef: spawn('child', { input: 69 })
+        }))
+      },
+      {
+        actors: {
+          child: {
+            src: createMachine({
+              context: ({ input }) => {
+                spy(input);
+                return {};
+              }
+            }),
+            input: 42
+          }
+        }
+      }
+    );
+
+    interpret(machine).start();
+
+    expect(spy).toHaveBeenCalledWith(69);
```

Comment:
Just to keep it clean 😅 
```suggestion
          childRef: spawn('child', { input: 100 })
        }))
      },
      {
        actors: {
          child: {
            src: createMachine({
              context: ({ input }) => {
                spy(input);
                return {};
              }
            }),
            input: 42
          }
        }
      }
    );

    interpret(machine).start();

    expect(spy).toHaveBeenCalledWith(100);
```